sunpy/map/mapbase.py

```diff
@@ -631,14 +631,16 @@ def observer_coordinate(self):
         """
         The Heliographic Stonyhurst Coordinate of the observer.
         """
+        missing_meta = {}
         for keys, kwargs in self._supported_observer_coordinates:
-            if all([k in self.meta for k in keys]):
+            meta_list = [k in self.meta for k in keys]
+            if all(meta_list):
                 return SkyCoord(obstime=self.date, **kwargs).heliographic_stonyhurst
-
-        all_keys = [str(e[0]) for e in self._supported_observer_coordinates]
-        all_keys = '\n'.join(all_keys)
-        warning_message = ("Missing metadata for observer: assuming Earth-based observer."
-                           "The following sets of keys were checked:\n" + all_keys)
+            elif any(meta_list) and not set(keys).isdisjoint(self.meta.keys()):
+                missing_meta[kwargs['frame']] = [key for key in set(keys).difference(self.meta.keys())]
```

Comment:
Do you need to cast to a list here? If you do, you can just do `list(set(keys).difference(self.meta.keys()))`

Reply:
Done.